src/components/interaction-designer/Connection.vue

```diff
@@ -217,24 +225,32 @@ export class Connection extends mixins(Lang) {
     this.activateConnection({connectionContext: this.connectionContext})
     this.activateBlock({blockId: null})
     this.$emit('lineMouseIn')
+    this.$emit('lineMouseClickedIn')
   }
 
-  clickAwayHandler(connectionElement: Element): void {
-    document.addEventListener('click', (event) => {
-      // Do not listen if the connection was not fully set
+  clickAwayHandler(connectionElement: Element, event: Event): void {
+    // Do not listen if the connection was not fully set
+    // TODO: Correctly implement the commented removeEventListener in case of performance issue on multiple bocks
```

Comment:
I've been trying to bring a performance improvement here, to remove the event listener, but couldn't succeed. I'm leaving a TODO just to help in case of need. Not an issue for now.